Lazy/LazyTests/Tests.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Threading;
+using Lazy;
+using NUnit.Framework;
+
+namespace LazyTests;
+
+public class Tests
+{
+    [Test]
+    public void TestForNullSupplier()
+    {
+        Assert.Throws<ArgumentNullException>(() => LazyFactory.CreateSingleThreadedLazy<int>(null));
+        Assert.Throws<ArgumentNullException>(() => LazyFactory.CreateMultiThreadedLazy<int>(null));
+    }
+
+    [Test]
+    public void TestForRepeatCalls()
+    {
+        var a = 1;
+        var function = LazyFactory.CreateSingleThreadedLazy<int>(()=> a);
+        while (a != 10)
+        {
+            Assert.IsTrue(function.Get() == 1);
```

Comment:
```suggestion
            Assert.AreEqual(1, function.Get());
```
Так оно будет более понятно ругаться